### PR TITLE
Fix cleanup in lifecycle node

### PR DESCRIPTION
**Pull request by aniketmpatil**
Cleanup for publisher and rs_driver pointers

### CODE REVIEW

**Review by lukeschmitt-tr — COMMENTED** [review]
Good catch. Are there any other pointer we'd need to reset? Are we reinitializing these properly in on_configure/on_activate?

File: src/trossen_rslidar_node.cpp [hunk 1]
```diff
@@ -252,6 +252,8 @@ CallbackReturn PointCloudLFNode::on_deactivate(const rclcpp_lifecycle::State & /
   RCLCPP_DEBUG(get_logger(), "Deactivating...");
 
   driver_ptr_->stop();
+  free_point_cloud_queue_.clear();
+  point_cloud_queue_.clear();
```
Comment (aniketmpatil) [inline src/trossen_rslidar_node.cpp:256]: I'm not sure how we can reset/reinitialize these on cleanup. These are class objects from a custom SyncQueue class inside the driver.

Comment (lukeschmitt-tr) [reply src/trossen_rslidar_node.cpp:256]: Do they get rebuilt in the driver setup during on_configure or on_activate?

Comment (aniketmpatil) [reply src/trossen_rslidar_node.cpp:256]: Queues are created in the constructor and cleared on cleanup

**Review by lukeschmitt-tr — APPROVED** [review]
Approved, assuming that this all works as intended.

**Comment by aniketmpatil** [comment]
Tested independently using a single lidar setup and tried activating and resetting the node using lifecycle manager service calls.

**Pull request merged**